docs/index.rst

```diff
@@ -1,51 +1,211 @@
+The Python client library for `CrateDB`_ implements the Python Database API
+Specification v2.0 (`PEP 249`_), and also includes the :ref:`CrateDB dialect
+<using-sqlalchemy>` for `SQLAlchemy`_.
 
-    Check out the `sample application`_ (and the corresponding `sample
-    application documentation`_) for a practical demonstration of this driver
-    in use.
+The Python driver can be used to connect to both `CrateDB OSS`_ and `CrateDB
+Cloud`_, and is verified to work on Linux, macOS, and Windows. It is used by
+the `Crash CLI`_, as well as other libraries and applications connecting to
+CrateDB from the Python ecosystem. It is verified to work with CPython, but
+it has also been tested successfully with `PyPy`_.
 
-    For general help using the Python Database API or SQLAlchemy, please consult
-    `PEP 249`_, the `SQLAlchemy tutorial`_, or the `SQLAlchemy documentation`_.
+Please make sure to also visit the section about :ref:`other-options`, using
+the :ref:`crate-reference:interface-postgresql` interface of `CrateDB`_.
 
-.. SEEALSO::
 
-   The CrateDB Python client library is an open source project and is `hosted
-   on GitHub`_.
+*************
+Documentation
+*************
 
-.. rubric:: Table of contents
+For general help about the Python Database API, or SQLAlchemy, please consult
+`PEP 249`_, the `SQLAlchemy tutorial`_, and the general `SQLAlchemy
+documentation`_.
+
+For more detailed information about how to install the client driver, how to
+connect to a CrateDB cluster, and how to run queries, consult the resources
+referenced below. A dedicated section demonstrates how to use the :ref:`blob
+storage capabilities <crate-reference:blob_support>` of CrateDB.
 
 .. toctree::
-   :maxdepth: 2
+    :titlesonly:
+
+    getting-started
+    connect
+    query
+    blobs
+
+
+DB API
+======
+
+Install package from PyPI.
+
+.. code-block:: shell
+
+    $ pip install crate
+
+Connect to CrateDB instance running on ``localhost``.
+
+.. code-block:: python
+
+    >>> # Connect using DB API.
+    >>> from crate import client
+    >>> with client.connect("localhost:4200", username="crate") as connection:
+    >>>     cursor = connection.cursor()
+    >>>     cursor.execute("SELECT * FROM sys.summits;")
+    >>>     print(cursor.fetchall())
+    >>>     cursor.close()
+
+Connect to `CrateDB Cloud`_.
+
+.. code-block:: python
+
+    >>> # Connect using DB API.
+    >>> from crate import client
+    >>> client.connect("https://admin:iM_n1%0xRzShsapU4U(lDGOj@example.aks1.westeurope.azure.cratedb.net:4200")
+
+
+SQLAlchemy
+==========
+
+The CrateDB dialect for `SQLAlchemy`_ offers convenient ORM access and supports
+CrateDB's ``OBJECT``, ``ARRAY``, and geospatial data types using `GeoJSON`_,
+supporting different kinds of `GeoJSON geometry objects`_.
+
+.. toctree::
+    :maxdepth: 2
+
+    sqlalchemy
+
+Install package from PyPI with DB API and SQLAlchemy support.
+
+.. code-block:: shell
+
+    $ pip install crate[sqlalchemy] pandas
```

Comment:
Depending on the shell, it might need to be wrapped in double quotes (for example when using zsh): `pip install "crate[sqlalchemy]" pandas`. This syntax is also used in `README.rst` (otherwise, installation will error with `zsh: no matches found: crate[sqlalchemy]`).

Reply:
Thanks for spotting 💯. @proddata told me he actually needed to use single quotes on zsh instead?

Reply:
Both single and double quotes work fine for me (zsh on macOS). Not sure if one is more correct than the other 🤷.

Reply:
I think it doesn't really matter in this case. Double quotes in theory allow command substitution. But I think single quotes are more "correct" in this case.

The problematic part in `zsh` is using square brackets without quotes, as those are used for globbing

Reply:
Thanks. Fixed with 8235a75be0.